app/Services/UserService.php

```diff
@@ -15,4 +18,36 @@ public function createUser(array $data)
         $user_data['password'] = Hash::make($data['password']);
         return User::create($user_data);
     }
+
+    public function resetPassword(array $data, string $token)
+    {
+        $userID = DB::table('users')->where('email', $data['email'])->value('id');
+        if ($userID != null) {
+            ResetPassword::updateOrCreate(
+                ['user_id' => $userID],
+                [
+                    'user_id' => $userID,
+                    'token' => $token
+                ]
+            );
+        } else {
+            return response()->json(['User not found!', 404]);
```

Comment:
service method shouldn't return 'response()'. If you want to return 404, do 'throw HttpNotFoundExc..', catch it in the controller, and return the response from the controller.